cli/cmd/trigger_sequence.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/keptn/keptn/cli/internal"
+	"github.com/keptn/keptn/cli/pkg/credentialmanager"
+	"github.com/keptn/keptn/cli/pkg/logging"
+	"github.com/spf13/cobra"
+)
+
+type sequenceStruct struct {
+	Project  *string `json:"project"`
+	Service  *string `json:"service"`
+	Stage    *string `json:"stage"`
+	Sequence *string `json:"sequence"`
+}
+
+var sequence = sequenceStruct{}
+
+// Shipyard can have multiple sequences with an arbitrary name (my-sequence-1, my-sequence-2) in a stage
+// The sequence name is used to identify the sequence to be triggered
+
+/*
+apiVersion: "spec.keptn.sh/0.2.2"
+kind: "Shipyard"
+metadata:
+  name: "shipyard-sockshop"
+spec:
+  stages:
+    - name: "dev"
+      sequences:
+        - name: "my-sequence-1"
+          tasks:
+            - name: "deployment"
+            - name: "test"
+            - name: "evaluation"
+            - name: "release"
+
+        - name: "my-sequence-2"
+          tasks:
+            - name: "deployment"
+            - name: "test"
+            - name: "evaluation"
+            - name: "release"
+*/
+
+var triggerSequenceCmd = &cobra.Command{
+	Use:     "sequence",
+	Aliases: []string{"sequence"},
+	Short:   "Triggers the execution of any sequence in a project",
+	Long: `Triggers the execution of any sequence in a project with an arbitrary name.
+The name of the sequence has to be provided as an argument to the command. The sequence name is used to identify the sequence to be triggered.
+`,
+	Example:      `keptn trigger sequence <sequence-name> --project=<project> --service=<service> --stage=<stage>`,
+	SilenceUsage: true,
+	Args: func(cmd *cobra.Command, args []string) error {
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return doTriggerSequence(sequence)
+	},
+}
+
+func doTriggerSequence(sequenceInputData sequenceStruct) error {
+	var endPoint url.URL
+	var apiToken string
+	var err error
+	if !mocking {
+		endPoint, apiToken, err = credentialmanager.NewCredentialManager(assumeYes).GetCreds(namespace)
+	} else {
+		endPointPtr, _ := url.Parse(os.Getenv("MOCK_SERVER"))
+		endPoint = *endPointPtr
+		apiToken = os.Getenv("MOCK_API_TOKEN")
+		// apiToken = ""
+	}
+	if err != nil {
+		return errors.New(authErrorMsg)
+	}
+
+	logging.PrintLog("Triggering sequence "+*sequenceInputData.Sequence+" in project "+*sequenceInputData.Project+" in stage "+*sequenceInputData.Stage+" service "+*sequenceInputData.Service, logging.InfoLevel)
+
+	api, err := internal.APIProvider(endPoint.String(), apiToken)
+	if err != nil {
+		return err
+	}
+
+	project, errObj := api.ProjectsV1().GetProject(apimodels.Project{ProjectName: *sequenceInputData.Project})
+	if errObj != nil {
+		return fmt.Errorf("error while retrieving information for project %v: %s", *sequenceInputData.Project, *errObj.Message)
+	}
+
+	// if no stage has been provided to the sequence command, use the first stage of the project
+	if sequenceInputData.Stage == nil || *sequenceInputData.Stage == "" {
```

Comment:
Is this made redundant by the PreChecks below where you inform a user that the stage must be provided?

Reply:
I think, no. This won't work. I have to remove this condition. 💯